autoload/test/gleam/gleamtest.vim

```diff
@@ -0,0 +1,39 @@
+if !exists('g:test#gleam#gleamtest#file_pattern')
+  let g:test#gleam#gleamtest#file_pattern = '\v\.gleam$'
+endif
+
+function! test#gleam#gleamtest#test_file(file) abort
+  return a:file =~# g:test#gleam#gleamtest#file_pattern
+endfunction
+
+function! test#gleam#gleamtest#build_position(type, position) abort
+  if a:type ==# 'nearest'
+    let name = s:nearest_test(a:position)
+    if empty(name)
+      return ['test', a:position['file']]
+    endif
+
+    return ['test', a:position['file'],  '--test-filter '.shellescape(name, 1)]
```

Comment:
Looking at the Gleam documentation, there doesn't appear to be any filtering options, is that correct? If so this can be simplified to the following

```suggestion
  if a:type ==# 'nearest'
    return ['test', a:position['file']]
```